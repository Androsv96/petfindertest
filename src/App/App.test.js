import React from 'react';
import { mount, shallow } from 'enzyme';
import MyComponent from './';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

const configureMockStore = configureStore();
configure({ adapter: new Adapter() });

const initialState = {
  petsReducer: {
    petsData: [],
    petsPagination: {},
    petsShowDetail: false,
    petSelected: {},
    showPetsProgressOn: false,
  },
  filtersReducer: {
    showFiltersSection: true,
    animalTypes: [],
    petsBreeds: [],
    selectedAnimalTypeFilter: "",
    showAnimalsTypesCollapsed: true,
    showANimalBreedsCollapsed: false,
    animalsBreedsShowFrom: 0,
    animalsBreedsShowUntil: 10,
    breedsSelected: [],
    currentPage: 1,
    showFiltersProgressOn: false,
  },
  utilitiesReducer: {
    hasInternetConnection: true,
  }
}

const store = configureMockStore(initialState);

it('Testing App component', () => {
  const wrapper = shallow(<MyComponent store={store} />).dive();
  expect(wrapper.getElements()).toMatchSnapshot();
})