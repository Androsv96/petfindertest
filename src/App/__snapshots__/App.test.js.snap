// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component 1`] = `
Array [
  <App
    dispatch={[Function]}
    filtersReducer={
      Object {
        "animalTypes": Array [],
        "animalsBreedsShowFrom": 0,
        "animalsBreedsShowUntil": 10,
        "breedsSelected": Array [],
        "currentPage": 1,
        "petsBreeds": Array [],
        "selectedAnimalTypeFilter": "",
        "showANimalBreedsCollapsed": false,
        "showAnimalsTypesCollapsed": true,
        "showFiltersProgressOn": false,
        "showFiltersSection": true,
      }
    }
    petsReducer={
      Object {
        "petSelected": Object {},
        "petsData": Array [],
        "petsPagination": Object {},
        "petsShowDetail": false,
        "showPetsProgressOn": false,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    utilitiesReducer={
      Object {
        "hasInternetConnection": true,
      }
    }
  />,
]
`;
